workers/src/data/extractors.rs

```diff
@@ -0,0 +1,73 @@
+//! Extract data from encoded payloads.
+//!
+//! FIXME: couldn't make it work with `alloy`+ABI, so
+//! using manual extracting for now.
+//! Re-check if it can work that way.
+
+use alloy::primitives::FixedBytes;
+use bytes::{Buf, BufMut, BytesMut};
+
+/// Minimum length of a packet.
+const MINIMUM_PACKET_LENGTH: usize = 93; // 1 + 8 + 4 + 32 + 4 + 32 + 32
+
+/// The whole header from the message.
+pub struct Header {
+    version: u8,
+    nonce: u64,
+    src_eid: u32,
+    sender_addr: u32,
```

Comment:
`sender_addr` is actually `address`, aligned to 32 bytes and padded with zero bytes on left (bytes are incoming with big-endian order)